### DISCUSSION TITLE
Plugin "Score Editor" offers no insertion dialog

### DISCUSSION THREAD

**Discussion by erwinhilb** [zim-desktop-wiki/zim-desktop-wiki#1923] [Q&A] [answered]
The manual for the plugin "Score Editor" to embed Lilypond-rendered musical scores into a zim page states, that a dialog should be available for insertion.
After proper activation of the plugin (and installed dependencies), I cannot find the way to access this dialog. The manual does not explain how to insert the musical scores.

I have tested on both Linux and Windows with Zim 0.74.3.
Any pointers would be appreciated.

**Comment by introt** [comment]
It should appear at the menu bar, at Insert > Score, similarly to the other Insert X -plugins. Do the other ones appear there?

**Comment by erwinhilb** [comment]
The other menus appears there but not Score.

 

 

Von: introt ***@***.***> 
Gesendet: Samstag, 12. Februar 2022 08:15
An: zim-desktop-wiki/zim-desktop-wiki ***@***.***>
Cc: erwinhilb ***@***.***>; Author ***@***.***>
Betreff: Re: [zim-desktop-wiki/zim-desktop-wiki] Plugin "Score Editor" offers no insertion dialog (Discussion #1923)

 

It should appear at the menu bar, at Insert > Score, similarly to the other Insert X -plugins. Do the other ones appear there?

—
Reply to this email directly, view it on GitHub <https://github.com/zim-desktop-wiki/zim-desktop-wiki/discussions/1923#discussioncomment-2162267> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/ACCK4THODS6WTAWH7SPE5VTU2YCHVANCNFSM5OGKOMWQ> .
Triage notifications on the go with GitHub Mobile for iOS <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>  or Android <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub> .

**Reply by introt** [reply]
> After proper activation of the plugin (and installed dependencies)

1. Just making sure - dependencies show up as OK? (shown below)
2. Are the plugin options as shown?
![image](https://user-images.githubusercontent.com/26069787/153710552-96a510c2-9f6d-47b0-8df0-7fe653793b06.png)
3. When you toggle the plugin off and on again, are the logs as shown here? Are there any ERRORs? If so, please post the full log, which you can find from `Tools > View debug log`
                
                DEBUG: Unloading plugin scoreeditor
                DEBUG: unregister_object: "image+score"
                DEBUG: unregister_object: "image+score"
                DEBUG: unregister_object: "image+score"
                DEBUG: unregister_object: "image+score"
                DEBUG: Loading description for plugin: scoreeditor
                DEBUG: Loading plugin: scoreeditor
                DEBUG: Load extension: <class 'zim.plugins.scoreeditor.BackwardScoreImageObjectType'>
                DEBUG: register_object: "image+score"
                DEBUG: Loading description for plugin: scoreeditor

4. What's your Linux distribution, and how did you install Zim & Lilypond?

Just tried it myself, should work on the latest release:

![image](https://user-images.githubusercontent.com/26069787/153710778-eff7099b-29cf-4fcd-aa45-a63f1465b6af.png)

![image](https://user-images.githubusercontent.com/26069787/153710855-62eb5b95-29e5-4f0e-bda4-84ba0a0034e5.png)
([example source](https://www.linuxjournal.com/article/8364))

**Comment by erwinhilb** [comment]
3. In Windows 10 the log shows:

 

DEBUG: Unloading plugin scoreeditor

DEBUG: unregister_object: "image+score"

INFO: Loading template from: A:\Programme\Zim Desktop Wiki\share\zim\templates\plugins\equationeditor.tex

ERROR: Exception in signal handler for changed on <zim.plugins.InsertedObjectTypeMap object at 0x00000253f49f5d00>

Traceback (most recent call last):

  File "build/venv/lib/python3.9/site-packages/zim/signals.py", line 367, in emit

  File "build/venv/lib/python3.9/site-packages/zim/gui/pageview/__init__.py", line 5714, in on_changed

AttributeError: 'InsertedObjectPageviewManager' object has no attribute 'insert_image_score'

DEBUG: unregister_object: "image+score"

DEBUG: unregister_object: "image+score"

DEBUG: unregister_object: "image+score"

DEBUG: Loading description for plugin: scoreeditor

DEBUG: Dialog response OK

DEBUG: Closed dialog "Einstellungen"

DEBUG: Action: show_preferences

DEBUG: Loading description for plugin: distractionfree

DEBUG: Opening dialog "Einstellungen"

DEBUG: Loading plugin: scoreeditor

DEBUG: Load extension: <class 'zim.plugins.scoreeditor.BackwardScoreImageObjectType'>

DEBUG: register_object: "image+score"

INFO: Loading template from: A:\Programme\Zim Desktop Wiki\share\zim\templates\plugins\equationeditor.tex

INFO: Remove file: C:\Users\Erwin\AppData\Local\Temp\zim-iarld_r9\equation.tex

DEBUG: Loading description for plugin: scoreeditor

DEBUG: Dialog response OK

DEBUG: Closed dialog "Einstellungen"

 

Von: introt ***@***.***> 
Gesendet: Samstag, 12. Februar 2022 13:15
An: zim-desktop-wiki/zim-desktop-wiki ***@***.***>
Cc: erwinhilb ***@***.***>; Author ***@***.***>
Betreff: Re: [zim-desktop-wiki/zim-desktop-wiki] Plugin "Score Editor" offers no insertion dialog (Discussion #1923)

 

After proper activation of the plugin (and installed dependencies)

1.	Just making sure - dependencies show up as OK? (shown below)
2.	Are the plugin options as shown?
 <https://user-images.githubusercontent.com/26069787/153710552-96a510c2-9f6d-47b0-8df0-7fe653793b06.png> 
3.	When you toggle the plugin off and on again, are the logs as shown here? Are there any ERRORs? If so, please post the full log, which you can find from Tools > View debug log

4.           DEBUG: Unloading plugin scoreeditor
5.           DEBUG: unregister_object: "image+score"
6.           DEBUG: unregister_object: "image+score"
7.           DEBUG: unregister_object: "image+score"
8.           DEBUG: unregister_object: "image+score"
9.           DEBUG: Loading description for plugin: scoreeditor
10.         DEBUG: Loading plugin: scoreeditor
11.         DEBUG: Load extension: <class 'zim.plugins.scoreeditor.BackwardScoreImageObjectType'>
12.         DEBUG: register_object: "image+score"
13.         DEBUG: Loading description for plugin: scoreeditor

14.	What's your Linux distribution, and how did you install Zim & Lilypond?

Just tried it myself, should work on the latest release:

 <https://user-images.githubusercontent.com/26069787/153710778-eff7099b-29cf-4fcd-aa45-a63f1465b6af.png> 

 <https://user-images.githubusercontent.com/26069787/153710855-62eb5b95-29e5-4f0e-bda4-84ba0a0034e5.png> 
(example source <https://www.linuxjournal.com/article/8364> )

—
Reply to this email directly, view it on GitHub <https://github.com/zim-desktop-wiki/zim-desktop-wiki/discussions/1923#discussioncomment-2163001> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/ACCK4TCQIHLC4NFGKAESY3LU2ZFNRANCNFSM5OGKOMWQ> .
Triage notifications on the go with GitHub Mobile for iOS <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>  or Android <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub> .

**Reply by introt** [reply]
Thanks, that sounds like a bug. Could you post the whole log as a [Gist](https://gist.github.com/)? I'd also like to see the logs you get on Linux, as well as your Linux distribution and how did you install Zim & Lilypond.

**Comment by erwinhilb** [comment]
In Manjaro - Linux

1. the dependencies looks ok. 

2. the plugin options are like yours but in German  

3. after toggle the plugin off and on again the log shows:

DEBUG: Loading plugin: scoreeditor
DEBUG: Load extension: <class 'zim.plugins.scoreeditor.BackwardScoreImageObjectType'>
DEBUG: register_object: "image+score"
WARNING: Buffer edited while page read-only - potential bug
WARNING: Buffer edited while textview read-only - potential bug
INFO: Loading template from: /usr/share/zim/templates/plugins/scoreeditor.ly
INFO: Running: ('lilypond', '--version') (cwd: None)
INFO: Loading template from: /usr/share/zim/templates/plugins/scoreeditor.ly
INFO: Running: ('lilypond', '--version') (cwd: None)
INFO: Loading template from: /usr/share/zim/templates/plugins/scoreeditor.ly
INFO: Running: ('lilypond', '--version') (cwd: None)
INFO: Loading template from: /usr/share/zim/templates/plugins/scoreeditor.ly
INFO: Running: ('lilypond', '--version') (cwd: None)
DEBUG: Loading description for plugin: scoreeditor

(zim:2142): Gtk-CRITICAL **: 18:02:58.407: gtk_container_foreach: assertion 'GTK_IS_CONTAINER (container)' failed

4. I'm using Manjaro Linux (archlinux) and I installed ZIM, lilypond and frescobaldi from the official package manager with "$ sudo pacman -S ..."

I’ll try to post the whole log as a Gist.

Thanks 

 

 

Von: introt ***@***.***> 
Gesendet: Sonntag, 13. Februar 2022 09:00
An: zim-desktop-wiki/zim-desktop-wiki ***@***.***>
Cc: erwinhilb ***@***.***>; Author ***@***.***>
Betreff: Re: [zim-desktop-wiki/zim-desktop-wiki] Plugin "Score Editor" offers no insertion dialog (Discussion #1923)

 

Thanks, that sounds like a bug. Could you post the whole log as a  <https://gist.github.com/> Gist? I'd also like to see the logs you get on Linux, as well as your Linux distribution and how did you install Zim & Lilypond.

—
Reply to this email directly,  <https://github.com/zim-desktop-wiki/zim-desktop-wiki/discussions/1923#discussioncomment-2166605> view it on GitHub, or  <https://github.com/notifications/unsubscribe-auth/ACCK4TBC34UZS3A72D65IE3U25QJJANCNFSM5OGKOMWQ> unsubscribe.
Triage notifications on the go with GitHub Mobile for  <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> iOS or  <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub> Android.

**Reply by introt** [reply]
> WARNING: Buffer edited while page read-only - potential bug
WARNING: Buffer edited while textview read-only - potential bug

Have you toggled the notebook editability off, as shown below?
![image](https://user-images.githubusercontent.com/26069787/153746388-48e31326-3c57-451e-9927-da1222d4887b.png)

(shouldn't affect Score not showing up in the Insert menu, but if you haven't, the log message shouldn't show up)

When you run `lilypond` in a terminal, which version does it report? There may have been breaking changes.

**Comment by erwinhilb** [comment]
https://gist.github.com/erwinhilb/4dcddf38bb0985d81eab59942e5afeb7

 

Von: introt ***@***.***> 
Gesendet: Sonntag, 13. Februar 2022 09:00
An: zim-desktop-wiki/zim-desktop-wiki ***@***.***>
Cc: erwinhilb ***@***.***>; Author ***@***.***>
Betreff: Re: [zim-desktop-wiki/zim-desktop-wiki] Plugin "Score Editor" offers no insertion dialog (Discussion #1923)

 

Thanks, that sounds like a bug. Could you post the whole log as a Gist <https://gist.github.com/> ? I'd also like to see the logs you get on Linux, as well as your Linux distribution and how did you install Zim & Lilypond.

—
Reply to this email directly, view it on GitHub <https://github.com/zim-desktop-wiki/zim-desktop-wiki/discussions/1923#discussioncomment-2166605> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/ACCK4TBC34UZS3A72D65IE3U25QJJANCNFSM5OGKOMWQ> .
Triage notifications on the go with GitHub Mobile for iOS <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>  or Android <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub> .

**Reply by introt** [reply]
Thanks! This log doesn't include the lines in your previous message though, could you post that one in its entirety too?

**Comment by introt** [comment]
I spun up a Manjaro Live session in a VM, installed Zim & lilypond and (after manually setting PYTHONPATH) got the following when trying to insert a score:

![image](https://user-images.githubusercontent.com/26069787/153751683-321c8649-d5d6-4ac2-85cf-52fedb5df8cf.png)

Also, the lilypond package doesn't seem to correctly pull its dependencies:

![image](https://user-images.githubusercontent.com/26069787/153751741-6ffa9abe-aa0c-4b7a-b79c-2203237b028c.png)

This might be due to the limitations of the live session, but I couldn't get a VM to boot after installation, so someone else needs to look into that... In any case, "Score" did appear in the Insert menu, so something's up with your installation @erwinhilb

**Comment by introt** [comment]
Newest Manjaro XFCE, Zim 0.74.3, Lilypond 2.22.1 should work, the menu item in Deutch is `Einfügen` > `Bewertung` (a wrong translation of "Score" for sure!). Do you see that option on your installation, @erwinhilb?

![image](https://user-images.githubusercontent.com/26069787/153757412-3263e27a-4a2c-4db4-888f-487ab05a31d5.png)

**Reply by dehesselle** [reply]
I've submitted a PR to fix that translation. https://github.com/zim-desktop-wiki/zim-desktop-wiki/pull/1928

**Comment by erwinhilb** [comment]
OK, „Bewertung“ is ther in insert menu and works fine in Linux.

In Windows when I want to see the preview I get this message:

 

This is zim 0.74.3

Platform: nt

Locale: de_DE cp1252

FS encoding: utf-8

Python: (3, 9, 7, 'final', 0)

PyGObject: (3, 42, 0)

 

======= Traceback =======

  File "build/venv/lib/python3.9/site-packages/zim/gui/insertedobjects.py", line 435, in _action_handler

  File "build/venv/lib/python3.9/site-packages/zim/plugins/base/imagegenerator.py", line 54, in new_model_interactive

  File "build/venv/lib/python3.9/site-packages/zim/plugins/base/imagegenerator.py", line 320, in run_dialog_for_model

  File "build/venv/lib/python3.9/site-packages/zim/plugins/base/imagegenerator.py", line 141, in set_from_generator

  File "build/venv/lib/python3.9/site-packages/zim/newfs/local.py", line 389, in write

TypeError: write() argument must be str, not None

 

Thanks

Erwin

 

Von: introt ***@***.***> 
Gesendet: Sonntag, 13. Februar 2022 15:26
An: zim-desktop-wiki/zim-desktop-wiki ***@***.***>
Cc: erwinhilb ***@***.***>; Mention ***@***.***>
Betreff: Re: [zim-desktop-wiki/zim-desktop-wiki] Plugin "Score Editor" offers no insertion dialog (Discussion #1923)

 

Newest Manjaro XFCE, Zim 0.74.3, Lilypond 2.22.1 should work, the menu item in Deutch is Einfügen > Bewertung (a wrong translation of "Score" for sure!). Do you see that option on your installation, @erwinhilb <https://github.com/erwinhilb> ?

 <https://user-images.githubusercontent.com/26069787/153757412-3263e27a-4a2c-4db4-888f-487ab05a31d5.png> 

—
Reply to this email directly, view it on GitHub <https://github.com/zim-desktop-wiki/zim-desktop-wiki/discussions/1923#discussioncomment-2167673> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/ACCK4TDWQPOVCW75GDXMQGDU265O5ANCNFSM5OGKOMWQ> .
Triage notifications on the go with GitHub Mobile for iOS <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>  or Android <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub> .

**Comment by erwinhilb** [comment]
Now in Windows using the insert > Bewertung :

 

This is zim 0.74.3

Platform: nt

Locale: de_DE cp1252

FS encoding: utf-8

Python: (3, 9, 7, 'final', 0)

PyGObject: (3, 42, 0)

 

======= Traceback =======

  File "build/venv/lib/python3.9/site-packages/zim/gui/insertedobjects.py", line 435, in _action_handler

  File "build/venv/lib/python3.9/site-packages/zim/plugins/base/imagegenerator.py", line 54, in new_model_interactive

  File "build/venv/lib/python3.9/site-packages/zim/plugins/base/imagegenerator.py", line 320, in run_dialog_for_model

  File "build/venv/lib/python3.9/site-packages/zim/plugins/base/imagegenerator.py", line 141, in set_from_generator

  File "build/venv/lib/python3.9/site-packages/zim/newfs/local.py", line 389, in write

TypeError: write() argument must be str, not None

 

 

Here the Gist with the whole zim.log in Windows

https://gist.github.com/erwinhilb/69fef90492cfe2be5aaf8a0f061f70a1

 

In Linux al is working ok now.

Thanks !

 

Von: introt ***@***.***> 
Gesendet: Sonntag, 13. Februar 2022 10:27
An: zim-desktop-wiki/zim-desktop-wiki ***@***.***>
Cc: erwinhilb ***@***.***>; Author ***@***.***>
Betreff: Re: [zim-desktop-wiki/zim-desktop-wiki] Plugin "Score Editor" offers no insertion dialog (Discussion #1923)

 

Thanks! This log doesn't include the lines in your previous message though, could you post that one in its entirety too?

—
Reply to this email directly, view it on GitHub <https://github.com/zim-desktop-wiki/zim-desktop-wiki/discussions/1923#discussioncomment-2166778> , or unsubscribe <https://github.com/notifications/unsubscribe-auth/ACCK4TELWYNY4NEIKIULKQLU252NBANCNFSM5OGKOMWQ> .
Triage notifications on the go with GitHub Mobile for iOS <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>  or Android <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub> .

**Reply by introt** [reply]
You're welcome! Thanks for the logs - seems like the lilypond executable can't be found:

> INFO: Running: ('convert-ly', '--current-version', '--edit', 'score.ly') (cwd: 'C:\\Users\\Erwin\\AppData\\Local\\Temp\\zim-1nty5176')
Traceback (most recent call last):
  File "build/venv/lib/python3.9/site-packages/zim/plugins/base/imagegenerator.py", line 352, in <lambda>
  File "build/venv/lib/python3.9/site-packages/zim/plugins/base/imagegenerator.py", line 383, in update_image
  File "build/venv/lib/python3.9/site-packages/zim/plugins/scoreeditor.py", line 126, in generate_image
  File "build/venv/lib/python3.9/site-packages/zim/applications.py", line 238, in run
  File "subprocess.py", line 951, in __init__
  File "subprocess.py", line 1420, in _execute_child
FileNotFoundError: [WinError 2] Das System kann die angegebene Datei nicht finden

For some reason, this fails silently, which seems to result in Zim trying to write `None` into the image file - producing the fail you're seeing in the GUI.

So - you need to have both `convert-ly` and `lilypond` in your path. Please follow the instructions for "Running on the command-line" at https://lilypond.org/windows.html, reboot, and make sure the executables can be found by opening a command prompt via pressing `Win` + `R` and entering `cmd`, and proceeding to execute `lilypond` and `convert-ly`.

Excuse my brevity, I have school in the morning :)

Edit: I've opened PR #1930 to make `convert-ly` a strict dependency

**Comment by erwinhilb** [comment, marked as answer]
Thanks a lot 🙂

Am 13. Februar 2022 21:18:31 schrieb introt ***@***.***>:

**Reply by introt** [reply]
You're welcome! I'll mark this thread as answered after the PRs get merged, feel free to open another one if you encounter any further issues.
Edit: I've opened #1946 to track this
